x-pack/filebeat/module/threatintel/malwarebazaar/test/malwarebazaar.ndjson.log-expected.json

```diff
@@ -416,5 +416,184 @@
+        "abusech.malwarebazaar.intelligence.uploads": 2,
+        "abusech.malwarebazaar.tags": [
+            "42 192 232 209",
+            "exe",
+            "signed"
+        ],
+        "event.category": "threat",
+        "event.dataset": "threatintel.malwarebazaar",
+        "event.kind": "enrichment",
+        "event.module": "threatintel",
+        "event.timezone": "-02:00",
+        "event.type": "indicator",
+        "fileset.name": "malwarebazaar",
+        "input.type": "log",
+        "log.offset": 9408,
+        "related.hash": [
+            "03970e27a3e5fbbffbbc434239f9f3bf",
+            "163ce73e6369b6830f7ee781052d2470",
+            "6144:x9ZURaJOfiCIorsoCv0h7E9cwFyNkkCIjq7O3IWU+MagLMNLeJFuxmXROnWnK6ia:x8Ry8ixorsoC8ZGVYd3nUrCfsXRzK6ia",
+            "678f39a7f5bc61c4231232a0fe44f49fea5349cd51f9c8808ee2b6602ff1d79",
+            "T196E48D06BED280FFD655193014BA67769A3A5B020B35CFC39398DF592D33171AE3A13A"
+        ],
+        "service.type": "threatintel",
+        "tags": [
+            "forwarded",
+            "threatintel-malwarebazaar"
+        ],
+        "threat.feed.dashboard_id": "ad9c7430-72de-11eb-a3e3-b3cc7c78a70f",
+        "threat.feed.name": "[Filebeat] AbuseCH MalwareBazaar",
+        "threat.indicator.file.extension": "exe",
+        "threat.indicator.file.hash.md5": "03970e27a3e5fbbffbbc434239f9f3bf",
+        "threat.indicator.file.hash.sha1": "bf8bf87788034e2312312312332031366b13db61",
+        "threat.indicator.file.hash.sha256": "678f39a7f5bc61c4231232a0fe44f49fea5349cd51f9c8808ee2b6602ff1d79",
+        "threat.indicator.file.hash.sha384": "6159d988d477309048cf7e540a23ed9a7b6b4c42b45e3e192ce08ddf8cd89cf4325a49146e80e8bc33b299fe36f81c83",
+        "threat.indicator.file.hash.ssdeep": "6144:x9ZURaJOfiCIorsoCv0h7E9cwFyNkkCIjq7O3IWU+MagLMNLeJFuxmXROnWnK6ia:x8Ry8ixorsoC8ZGVYd3nUrCfsXRzK6ia",
+        "threat.indicator.file.hash.tlsh": "T196E48D06BED280FFD655193014BA67769A3A5B020B35CFC39398DF592D33171AE3A13A",
+        "threat.indicator.file.mime_type": "application/x-dosexec",
+        "threat.indicator.file.name": "678f39a7f5bc61c4c9a0f34a0fe44f49fea5349cd51211231234153211ff1d79",
+        "threat.indicator.file.pe.imphash": "163ce73e6369b6830f7ee781052d2470",
+        "threat.indicator.file.size": 688472,
+        "threat.indicator.file.x509.issuer.common_name": [
+            "Apple Worldwide Developer Relations Certification Authority",
+            "Lenovo ThinkCentre X81 12DT003FUC"
+        ],
+        "threat.indicator.file.x509.not_after": [
+            "2024-05-06T06:07:31Z",
+            "2032-04-01T14:03:55Z"
+        ],
+        "threat.indicator.file.x509.not_before": [
+            "2021-05-07T06:07:32Z",
+            "2022-03-31T14:03:55Z"
+        ],
+        "threat.indicator.file.x509.public_key_algorithm": [
+            "sha1WithRSAEncryption",
+            "sha256WithRSAEncryption"
+        ],
+        "threat.indicator.file.x509.serial_number": [
+            "",
```

Comment:
This empty string is here to test that the processor handles the unlikely cases where the code signing is missing fields. I attempted to conditionally not append when `_ingest?._value?.<fieldname> != ""` but this invariably broke the pipeline with timeouts (using a variety of candidate syntaxes).

Reply:
Unfortunately, it's not possible to access inside a condition `_ingest._value` as it is not visible to painless.

The only alternative I can suggest is to use a script processor to split the array of objects into separate array fields. But if we're not expecting missing fields then this approach is good

Reply:
For illustrative purposes:

```
  - script:
      lang: painless
      if: 'ctx.abusech?.malwarebazaar?.code_sign != null'
      source: |
        Map dest = new HashMap();
        for (def obj : ctx.abusech.malwarebazaar.code_sign) {
          for (def entry : obj.entrySet()) {
            dest.computeIfAbsent(entry.getKey(), k -> new ArrayList()).add(entry.getValue());
          }
        }
        ctx["_temp_"] = dest;

  - rename:
      field: _temp_.subject_cn
      target_field: threat.indicator.file.x509.subject.common_name
      ignore_missing: true
  - rename:
      field: _temp_.issuer_cn
      target_field: threat.indicator.file.x509.issuer.common_name
      ignore_missing: true
  - rename:
      field: _temp_.algorithm
      target_field: threat.indicator.file.x509.public_key_algorithm
      ignore_missing: true
  - rename:
      field: _temp_.valid_from
      target_field: threat.indicator.file.x509.not_before
      ignore_missing: true
  - rename:
      field: _temp_.valid_to
      target_field: threat.indicator.file.x509.not_after
      ignore_missing: true
  - rename:
      field: _temp_.serial_number
      target_field: threat.indicator.file.x509.serial_number
      ignore_missing: true
```

Reply:
Thanks. We can revisit this if we find that empty fields ever find their way into the `code_sign` array.